packages/governance/src/multisig/multisig.cairo

```diff
@@ -0,0 +1,503 @@
+                    let signer_to_remove = *signer;
+                    if !Multisig::is_signer(@self, signer_to_remove) {
+                        continue;
+                    }
+                    let last_index = signers_count - 1;
+                    let index = self.Multisig_signers_indices.read(signer_to_remove);
+                    if index != last_index {
+                        // Swap signer to remove with the last signer
+                        let last_signer = self.Multisig_signers_by_index.read(last_index);
+                        self.Multisig_signers_indices.write(last_signer, index);
+                        self.Multisig_signers_by_index.write(index, last_signer);
+                    }
+                    // Remove the last signer
+                    self.Multisig_is_signer.write(signer_to_remove, false);
+                    self.Multisig_signers_by_index.write(last_index, Zero::zero());
+                    self.Multisig_signers_indices.write(signer_to_remove, 0);
+
+                    self.emit(SignerRemoved { signer: signer_to_remove });
+
+                    signers_count -= 1;
+                };
+                self.Multisig_signers_info.write(SignersInfo { quorum, signers_count });
+            }
+            self._change_quorum(new_quorum);
+        }
+
+        fn _replace_signer(
+            ref self: ComponentState<TContractState>,
+            signer_to_remove: ContractAddress,
+            signer_to_add: ContractAddress
+        ) {
+            assert(signer_to_add.is_non_zero(), Errors::ZERO_ADDRESS);
+            assert(!Multisig::is_signer(@self, signer_to_add), Errors::ALREADY_A_SIGNER);
+            assert(Multisig::is_signer(@self, signer_to_remove), Errors::NOT_A_SIGNER);
+
+            self.Multisig_is_signer.write(signer_to_remove, false);
+            self.Multisig_is_signer.write(signer_to_add, true);
+            let index = self.Multisig_signers_indices.read(signer_to_remove);
+            self.Multisig_signers_by_index.write(index, signer_to_add);
+            self.Multisig_signers_indices.write(signer_to_add, index);
+            self.Multisig_signers_indices.write(signer_to_remove, 0);
+
+            self.emit(SignerRemoved { signer: signer_to_remove });
+            self.emit(SignerAdded { signer: signer_to_add });
+        }
+
+        fn _change_quorum(ref self: ComponentState<TContractState>, new_quorum: u32) {
+            let SignersInfo { quorum: old_quorum, signers_count } = self
+                .Multisig_signers_info
+                .read();
+            if new_quorum != old_quorum {
+                assert(new_quorum.is_non_zero(), Errors::ZERO_QUORUM);
+                assert(new_quorum <= signers_count, Errors::QUORUM_TOO_HIGH);
+                self.Multisig_signers_info.write(SignersInfo { quorum: new_quorum, signers_count });
+                self.emit(QuorumUpdated { old_quorum, new_quorum });
+            }
+        }
+    }
+}
+
+//
+// Storage helpers
```

Comment:
Let's move this helpers into a different file, something like `storage_utils.cairo` under the same directory.

Reply:
Moved to a separate file